src/connector/src/source/kinesis/source/reader.rs

```diff
@@ -22,32 +21,22 @@ use aws_sdk_kinesis::model::ShardIteratorType;
 use aws_sdk_kinesis::output::GetRecordsOutput;
 use aws_sdk_kinesis::types::SdkError;
 use aws_sdk_kinesis::Client as KinesisClient;
-use futures::future::join_all;
-use futures_async_stream::{for_await, try_stream};
-use futures_concurrency::prelude::*;
-use tokio::sync::Mutex;
-use tokio::task::JoinHandle;
+use futures::stream::FuturesUnordered;
+use futures::TryStreamExt;
+use futures_async_stream::try_stream;
+use risingwave_common::util::select_all;
 
 use crate::source::kinesis::source::message::KinesisMessage;
 use crate::source::kinesis::split::{KinesisOffset, KinesisSplit};
 use crate::source::kinesis::{build_client, KinesisProperties};
-use crate::source::{Column, ConnectorState, SourceMessage, SplitId, SplitImpl, SplitReader};
+use crate::source::{
+    BoxSourceStream, Column, ConnectorState, SourceMessage, SplitId, SplitImpl, SplitReader,
+};
 
 pub struct KinesisMultiSplitReader {
     /// splits are not allowed to be empty, otherwise connector source should create
     /// DummySplitReader which is always idling.
-    splits: Vec<KinesisSplit>,
-    properties: KinesisProperties,
-    message_cache: Arc<Mutex<Vec<SourceMessage>>>,
-    consumer_handler: Option<JoinHandle<()>>,
-}
-
-impl Drop for KinesisMultiSplitReader {
-    fn drop(&mut self) {
-        if let Some(handler) = self.consumer_handler.as_mut() {
-            handler.abort();
-        }
-    }
+    readers: Vec<KinesisSplitReader>,
 }
```

Comment:
please remove the struct. It was for consuming multiple splits and now we do it in the source layer

Reply:
done in 6fd08e3. PTAL!